ARCore/app/src/main/java/com/novoda/spikes/arcore/ModelCollection.kt

```diff
@@ -0,0 +1,54 @@
+package com.novoda.spikes.arcore
+
+import android.util.Log
+import com.novoda.spikes.arcore.poly.PolyAsset
+import com.novoda.spikes.arcore.poly.PolyAssetLoader
+import com.novoda.spikes.arcore.rendering.PolyAssetRenderer
+
+class ModelCollection(
+        private val assetLoader: PolyAssetLoader,
+        private val messageDisplayer: MessageDisplayer,
+        private val modelsDisplayer: ModelsDisplayer
+) : ARClassifier.Listener {
+
+    private val models = HashMap<PolyAsset, PolyAssetRenderer>()
+
+    private val assets : List<PolyAsset>
+        get() = models.keys.toList()
+    
+    var currentModel: PolyAssetRenderer? = null
+
+    private var modelModulo = 0
+
+    override fun onObjectClassified(label: String) {
+        if (wasNotAlreadySearchedFor(label) && modelModulo == 0) { //We start with a basic modulo, we could improve with a reliability based load
```

Comment:
We only query the API every 30 classifications  with the last classified label. We could improve this by running a total or using the confidence value back from tensorflow. A better version would query for the most likely object once it's confident enough maybe ?